### DISCUSSION TITLE
Logging client disconnection exceptions in EventSourceResponse

### DISCUSSION THREAD

**Discussion by yonaka15** [fastapi/fastapi#10340] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from sse_starlette.sse import EventSourceResponse

import asyncio

app = FastAPI()

async def event_generator():
    while True:
        yield f"data: Some event data\n\n"
        await asyncio.sleep(1)

@app.get("/events/")
async def get_event():
    return EventSourceResponse(event_generator())
```

### Description

When a client parts ways during an EventSourceResponse stream, an exception pops up. It would be great if FastAPI's logger could automatically capture this, making it easier for us to track what's happening.

Suggested Fix:
How about we set the logger to automatically record these exceptions and label them with an attention-grabbing level, like "WARNING"? What do you think?

### Operating System

macOS

### FastAPI Version

0.97.0

### Pydantic Version

1.10.4

### Python Version

3.9

**Comment by n8sty** [comment, marked as answer]
FastAPI doesn't too much logging itself beyond some warnings about missing or incorrect installations so I don't think this will get adopted. You can probably achieve what you want by using a custom exception handler as seen here https://fastapi.tiangolo.com/tutorial/handling-errors/#install-custom-exception-handlers.

**Comment by yonaka15** [comment]
Thank you for the valuable advice. I will refer to the link provided.
